OlawaleJoseph/E-Vote-Backend
# Ft get user polls

# What does this PR do?
This PR implements the functionality to get multiple or single poll of the logged-in user

## Implementations
- Authenticate user
- Validate poll id param for getting single poll
- Ensure a user can only see his/her poll/polls

## Testing
  - Clone this app by running the command `git clone git@github.com:OlawaleJoseph/E-Vote-Backend.git` in a terminal
  - cd into the created directory/folder, open a terminal in the root of the directory and run the command `git checkout ft-get-user-polls`
  - Create a PostgreSQL database on your machine, navigate to the config/dtabase.yml file in the directory, update the database.yml to reflect your database configurations for development and test environments.
  - run `bundle install` to install all dependencies
  - run `rspec`

 [Check here to learn how to create a poll/polls](https://github.com/OlawaleJoseph/E-Vote-Backend/pull/5)

### Getting of polls
- Login as a user and copy the token in the `Authorization` header
### To get all polls
- Make a GET request to `http://localhost:3000/api/v1/polls` with the token from the previous step in the request header as `Authorization`

### To get a single poll
- Make a GET request to `http://localhost:3000/api/v1/polls/{poll_id}` with the token from the previous step in the request header as `Authorization`
